src/sidebar/services/router.js

```diff
@@ -94,9 +98,15 @@ export class RouterService {
         throw new Error(`Cannot generate URL for route "${name}"`);
     }
 
-    const query = queryString.stringify(queryParams);
-    if (query.length > 0) {
-      url += '?' + query;
+    let hasParams = false;
+    const searchParams = new URLSearchParams();
+    for (let [key, value] of Object.entries(queryParams)) {
+      hasParams = true;
```

Comment:
Lazy question: is there a test for the no-params path? I presume so if coverage is happy.

Reply:
Yes. The tests for the `navigate` method include cases both with and without params in the URL.